bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -393,12 +415,19 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
   }
 
   if (readOnly) {
-    executeReadOnlyRequest(span, m);
-    delete m;
+    if (activeExecutions_ > 0) {
+      deferredRORequests_.push_back(
```

Comment:
What is the problem with handling read-only requests during the execution of other requests?
In the worst case, the reply won't be consistent with the latest updates in the db

Reply:
Currently, we are trying to do this change as much conservatively we can and to keep things similar to the current situation.

Reply:
After discussion with Guy, he also said that it's dangerous to execute RO requests at the same time of post-execution since the state is changing and it can cause reading wrong state and at the end replica collapse.